### PR TITLE
[ASCII-2077] upgrade golangci-lint to a version that support module plugin

### PR DESCRIPTION
**Pull request by GustavoCaso**
### What does this PR do?

Upgrade golangci-lint to version 1.57.0 https://golangci-lint.run/product/changelog/#v1570

This is part of the effort of adding custom Go lang linters. We need to use a version that supports the new [module plugin](https://golangci-lint.run/plugins/module-plugins/) for writing custom linters.

If you want to have a look at a PoC that uses the module plugin system, you can! https://github.com/DataDog/datadog-agent/pull/27033

I fixed the revive warnings using custom script that uses the JSON output of `golanci-lint`  and `revive` https://github.com/DataDog/experimental/pull/3532

### CODE REVIEW

**Review by FlorentClarret — APPROVED** [review]

File: cmd/cluster-agent/api/agent/agent.go [hunk 1]
```diff
@@ -162,7 +162,7 @@ func makeFlare(w http.ResponseWriter, r *http.Request, statusComponent status.Co
 }
 
 //nolint:revive // TODO(CINT) Fix revive linter
-func getConfigCheck(w http.ResponseWriter, r *http.Request, ac autodiscovery.Component) {
+func getConfigCheck(w http.ResponseWriter, _ *http.Request, ac autodiscovery.Component) {
```
Comment (davidor) [inline cmd/cluster-agent/api/agent/agent.go:165]: We should remove the TODO here

File: comp/core/autodiscovery/autodiscoveryimpl/autoconfig_test.go [hunk 2]
```diff
@@ -46,7 +46,7 @@ type MockProvider struct {
 }
 
 //nolint:revive // TODO(AML) Fix revive linter
-func (p *MockProvider) Collect(ctx context.Context) ([]integration.Config, error) {
+func (p *MockProvider) Collect(_ context.Context) ([]integration.Config, error) {
```
Comment (davidor) [inline comp/core/autodiscovery/autodiscoveryimpl/autoconfig_test.go:49]: Delete the TODO and also in the other 2 occurrences of this file

File: comp/core/autodiscovery/autodiscoveryimpl/common_test.go [hunk 3]
```diff
@@ -72,14 +72,14 @@ func (s *dummyService) IsReady(context.Context) bool {
 // HasFilter returns false
 //
 //nolint:revive // TODO(AML) Fix revive linter
-func (s *dummyService) HasFilter(filter containers.FilterType) bool {
+func (s *dummyService) HasFilter(_ containers.FilterType) bool {
```
Comment (davidor) [inline comp/core/autodiscovery/autodiscoveryimpl/common_test.go:75]: Delete TODO

**Review by davidor — COMMENTED** [review]
From the container-platform files, changes look good because it's mostly replacing unused vars with "_". However, there are many TODOs that should be deleted, I only pointed out a few.

**Review by ogaca-dd — APPROVED** [review]

**Pull request merged**